OrbitQt/ProfilingTargetDialog.cpp

```diff
@@ -436,38 +406,51 @@ void ProfilingTargetDialog::SelectFile() {
   }
 }
 
-void ProfilingTargetDialog::TrySelectProcess(const ProcessData& process) {
+bool ProfilingTargetDialog::TrySelectProcess(const ProcessData& process) {
   QModelIndexList matches = process_proxy_model_.match(
       process_proxy_model_.index(0, static_cast<int>(ProcessItemModel::Column::kName)),
       Qt::DisplayRole, QVariant::fromValue(QString::fromStdString(process.name())));
 
-  if (matches.isEmpty()) return;
+  if (matches.isEmpty()) return false;
 
   LOG("Selecting remembered process: %s", process.name());
 
   ui_->processesTableView->selectionModel()->setCurrentIndex(
       matches[0], {QItemSelectionModel::SelectCurrent, QItemSelectionModel::Rows});
+  return true;
 }
 
 void ProfilingTargetDialog::OnProcessListUpdate(ProcessManager* process_manager) {
-  main_thread_executor_->Schedule([this, process_manager]() {
-    // When this function call is scheduled, the process_manager might not exist anymore
-    if (process_manager == nullptr) return;
-
-    bool had_processes_before = process_model_.HasProcesses();
-    process_model_.SetProcesses(process_manager->GetProcessList());
-    if (!had_processes_before) return;
-
-    if (ui_->processesTableView->selectionModel()->hasSelection()) return;
-
-    if (process_ != nullptr) {
-      TrySelectProcess(*process_);
-    }
-
-    if (ui_->processesTableView->selectionModel()->hasSelection()) return;
-
-    ui_->processesTableView->selectRow(0);
-  });
+  std::vector<ProcessInfo> process_list = process_manager->GetProcessList();
+
+  QMetaObject::invokeMethod(QCoreApplication::instance(),
```

Comment:
This makes me wonder: Why `QCoreApplication::instance()`? Does `this` not work? Does that mean `this` and `QCoreApplication::instance()` have different threads associated?